lib/hl7/message.ex

```diff
@@ -508,18 +508,15 @@ defmodule HL7.Message do
     end
   end
 
-  defp validate_text(raw_text, options) do
-    encoded_text =
-      if options[:accept_latin1] == true do
-        :unicode.characters_to_binary(raw_text, :latin1)
-      else
-        raw_text
-      end
+  defp validate_text(raw_text, %{accept_latin1: true}) do
+    {:ok, transcode(raw_text)}
```

Comment:
Per last change, just ripped out the validation for this case. It gets validated per the transform implicitly. I can not think of any performance reason why we would want to do essentially the same operation twice. It's not exactly twice (i.e., String.valid?/1 will bail early on), but for any binary where String.valid?/1 returns true, we'll pay the toll twice.